Test notebooks with nbval

@mikemhenry volunteered to help us out with testing our notebooks with `nbval` on PR #7. We should try and get that set up to confirm that our notebooks are running correctly. Such testing might also help us catch issues with our Binder environments if we can run the tests on CI with the appropriate environment created, but I'm not familiar enough with `nbval` to know how difficult that might be.

Okay this is the first bug I hit:
```
================================================================================================= FAILURES =================================================================================================
_______________________________________________________________ examples/Using Machine Learning for Structural Identification.ipynb::Cell 9 ________________________________________________________________
Notebook cell execution failed
Cell 9: Cell execution caused an exception

Input:
from umap import UMAP
umap = UMAP(random_state=42)

X_reduced = umap.fit_transform(X)

Traceback:

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-10-3c8472543e01> in <module>
----> 1 from umap import UMAP
      2 umap = UMAP(random_state=42)
      3 
      4 X_reduced = umap.fit_transform(X)

ImportError: cannot import name 'UMAP' from 'umap' (/Users/mike/miniconda3/envs/freud/lib/python3.7/site-packages/umap/__init__.py)
```

With the most recent freud + this repo I'm running into this error
```

================================================================================================= FAILURES =================================================================================================
_________________________________________________________________________________ module_intros/pmft.PMFTXYZ.ipynb::Cell 4 _________________________________________________________________________________
Notebook cell execution failed
Cell 4: Cell execution caused an exception

Input:
shiftvec = [0.82, 0.82, 0]

window = 2**(1/6)/6 # Smaller window for the shifted case

bins = 50

pmft = freud.pmft.PMFTXYZ(x_max=window, y_max=window, z_max=window,
                          bins=50, shiftvec=shiftvec)

# This data is for isotropic particles, so we will just make some unit quaternions
# to use as the orientations
orientations = np.array([[1, 0, 0, 0]] * pos_data.shape[1])

for i in range(10, pos_data.shape[0]):
    box = box_data[i]
    points = pos_data[i]
    pmft.compute(system=(box, points), orientations=orientations, reset=False)

shifted = pmft.pmft

x, y, z = pmft.bin_centers

Traceback:

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-31e093ff8424> in <module>
     15     box = box_data[i]
     16     points = pos_data[i]
---> 17     pmft.compute(system=(box, points), orientations=orientations, reset=False)
     18 
     19 shifted = pmft.pmft

~/Projects/freud/freud/pmft.pyx in freud.pmft.PMFTXYZ.compute()

TypeError: compute() takes at least 2 positional arguments (1 given)
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -23,10 +23,15 @@ references:
     run:
       name: Load conda environment
       command: |
+        # Create the mybinder environment
         conda env create -f environment.yml
         conda activate freud-examples
+        # Remove freud so it can be built from source
         conda uninstall freud
-        conda install gcc_linux-64 gxx_linux-64 cython numpy tbb tbb-devel pytest nbval
+        # Install build requirements for freud
+        conda install gcc_linux-64 gxx_linux-64 cython numpy tbb tbb-devel
+        # Install testing requirements
+        conda install pytest nbval
 
   build_freud: &build_freud
     run:

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -4,7 +4,7 @@ references:
   container_miniconda: &container_miniconda
     docker:
       - image: continuumio/miniconda
-    working_directory: /opt
+    working_directory: ~/ci/freud-examples
 
   load_code: &load_code
     checkout
@@ -22,7 +22,7 @@ references:
     run:
       name: Build freud from master branch
       command: |
-        cd /opt
+        cd ~/ci/
         git clone https://github.com/glotzerlab/freud --recurse-submodules
         cd freud
         rm freud/*.cpp

--- a/environment.yml
+++ b/environment.yml
@@ -4,12 +4,16 @@ channels:
   - defaults
 dependencies:
   - _libgcc_mutex=0.1=main
+  - biopython=1.74=py37h516909a_0
   - bokeh=1.3.4=py37_0
   - bzip2=1.0.8=h516909a_1
   - ca-certificates=2019.9.11=hecc5488_0
   - certifi=2019.9.11=py37_0
+  - cftime=1.0.4.2=py37hc1659b7_0
+  - curl=7.65.3=hf8cf82a_0
   - cycler=0.10.0=py_2
   - dbus=1.13.6=he372182_0
+  - decorator=4.4.1=py_0
   - embree3=3.6.1=hc9558a2_0
   - expat=2.2.5=he1b5a44_1004
   - fftw=3.3.8=nompi_h7f3a6c3_1110
@@ -19,18 +23,25 @@ dependencies:
   - freud=2.0.0=py37hc9558a2_0
   - gettext=0.19.8.1=hc5be6a0_1002
   - glib=2.58.3=h6f030ca_1002
+  - griddataformats=0.5.0=py_0
+  - gsd=1.9.3=py37hc1659b7_0
   - gst-plugins-base=1.14.5=h0935bb2_0
   - gstreamer=1.14.5=h36ae1b5_0
+  - hdf4=4.2.13=h9a582f1_1002
+  - hdf5=1.10.5=nompi_h3c11f04_1104
   - hoomd=2.8.0=py37h9de70de_0
   - icu=64.2=he1b5a44_1
   - jinja2=2.10.3=py_0
   - joblib=0.14.0=py_0
   - jpeg=9c=h14c3975_1001
   - kiwisolver=1.1.0=py37hc9558a2_0
+  - krb5=1.16.3=h05b26f9_1001
   - lammps=2019.08.07=py37_openmpi_0
   - libblas=3.8.0=14_openblas
   - libcblas=3.8.0=14_openblas
   - libclang=9.0.0=hc9558a2_1
+  - libcurl=7.65.3=hda55be3_0
+  - libedit=3.1.20170329=hf8c457e_1001
   - libffi=3.2.1=he1b5a44_1006
   - libgcc-ng=9.1.0=hdf63c60_0
   - libgfortran-ng=7.3.0=hdf63c60_2
@@ -38,8 +49,10 @@ dependencies:
   - liblapack=3.8.0=14_openblas
   - libllvm8=8.0.1=hc9558a2_0
   - libllvm9=9.0.0=hc9558a2_3
+  - libnetcdf=4.7.1=nompi_h94020b1_101
   - libopenblas=0.3.7=h6e990d7_3
   - libpng=1.6.37=hed695b0_0
+  - libssh2=1.8.2=h22169c7_2
   - libstdcxx-ng=9.1.0=hdf63c60_0
   - libtiff=4.1.0=hfc65ed5_0
   - libuuid=2.32.1=h14c3975_1000
@@ -51,8 +64,13 @@ dependencies:
   - markupsafe=1.1.1=py37h516909a_0
   - matplotlib=3.1.2=py37_0
   - matplotlib-base=3.1.2=py37h250f245_0
+  - mdanalysis=0.20.1=py37he1b5a44_0
+  - mmtf-python=1.1.2=py_0
   - mpi=1.0=openmpi
+  - msgpack-python=0.6.2=py37hc9558a2_0
   - ncurses=6.1=hf484d3e_1002
+  - netcdf4=1.5.3=py37hf3213d6_0
+  - networkx=2.4=py_0
   - nspr=4.23=he1b5a44_0
   - nss=3.47=he751ad9_0
   - numba=0.46.0=py37hb3f55d8_1
@@ -61,6 +79,8 @@ dependencies:
   - openmpi=4.0.1=hc99cbb1_2
   - openssl=1.1.1d=h516909a_0
   - packaging=19.2=py_0
+  - pandas=0.25.3=py37hb3f55d8_0
+  - patsy=0.5.1=py_0
   - pcre=8.43=he1b5a44_0
   - pillow=6.2.1=py37h6b7be26_0
   - pip=19.3.1=py37_0
@@ -69,6 +89,7 @@ dependencies:
   - pyqt=5.12.3=py37hcca6a23_0
   - python=3.7.3=h33d41f4_1
   - python-dateutil=2.8.1=py_0
+  - pytz=2019.3=py_0
   - pyyaml=5.1.2=py37h516909a_0
   - qhull=2015.2=h6bb024c_1000
   - qt=5.12.5=hd8c4c69_1
@@ -76,12 +97,15 @@ dependencies:
   - rowan=1.2.2=py_2
   - scikit-learn=0.21.3=py37hcdab131_0
   - scipy=1.3.1=py37h921218d_2
+  - seaborn=0.9.0=py_1
   - setuptools=41.6.0=py37_1
   - six=1.13.0=py37_0
   - sqlite=3.30.1=hcee41ef_0
+  - statsmodels=0.10.1=py37hc1659b7_1
   - tbb=2019.9=hc9558a2_0
   - tk=8.6.9=hed695b0_1003
   - tornado=6.0.3=py37h516909a_0
+  - tqdm=4.36.1=py_0
   - umap-learn=0.3.10=py37_0
   - wheel=0.33.6=py37_0
   - xorg-libxau=1.0.9=h14c3975_0
@@ -91,5 +115,6 @@ dependencies:
   - zlib=1.2.11=h516909a_1006
   - zstd=1.4.3=h3b9ef0a_0
   - pip:
+    - plato-draw==1.7.0
     - pyqt5-sip==4.19.18
     - pyqtwebengine==5.12.1

--- a/environment.yml
+++ b/environment.yml
@@ -4,7 +4,10 @@ channels:
   - defaults
 dependencies:
   - _libgcc_mutex=0.1=main
+  - attrs=19.3.0=py_0
+  - backcall=0.1.0=py_0
   - biopython=1.74=py37h516909a_0
+  - bleach=3.1.0=py_0
   - bokeh=1.3.4=py37_0
   - bzip2=1.0.8=h516909a_1
   - ca-certificates=2019.9.11=hecc5488_0
@@ -14,7 +17,9 @@ dependencies:
   - cycler=0.10.0=py_2
   - dbus=1.13.6=he372182_0
   - decorator=4.4.1=py_0
+  - defusedxml=0.6.0=py_0
   - embree3=3.6.1=hc9558a2_0
+  - entrypoints=0.3=py37_1000
   - expat=2.2.5=he1b5a44_1004
   - fftw=3.3.8=nompi_h7f3a6c3_1110
   - fontconfig=2.13.1=h86ecdb6_1001
@@ -31,15 +36,26 @@ dependencies:
   - hdf5=1.10.5=nompi_h3c11f04_1104
   - hoomd=2.8.0=py37h9de70de_0
   - icu=64.2=he1b5a44_1
+  - importlib_metadata=0.23=py37_0
+  - ipydatawidgets=4.0.1=py_0
+  - ipykernel=5.1.3=py37h5ca1d4c_0
+  - ipyscales=0.3.0=py_0
+  - ipython=7.9.0=py37h5ca1d4c_0
+  - ipython_genutils=0.2.0=py_1
+  - ipywidgets=7.5.1=py_0
+  - jedi=0.15.1=py37_0
   - jinja2=2.10.3=py_0
   - joblib=0.14.0=py_0
   - jpeg=9c=h14c3975_1001
+  - jsonschema=3.1.1=py37_0
+  - jupyter_client=5.3.3=py37_1
+  - jupyter_core=4.5.0=py_0
   - kiwisolver=1.1.0=py37hc9558a2_0
   - krb5=1.16.3=h05b26f9_1001
   - lammps=2019.08.07=py37_openmpi_0
   - libblas=3.8.0=14_openblas
   - libcblas=3.8.0=14_openblas
-  - libclang=9.0.0=hc9558a2_1
+  - libclang=9.0.0=hc9558a2_2
   - libcurl=7.65.3=hda55be3_0
   - libedit=3.1.20170329=hf8c457e_1001
   - libffi=3.2.1=he1b5a44_1006
@@ -52,6 +68,7 @@ dependencies:
   - libnetcdf=4.7.1=nompi_h94020b1_101
   - libopenblas=0.3.7=h6e990d7_3
   - libpng=1.6.37=hed695b0_0
+  - libsodium=1.0.17=h516909a_0
   - libssh2=1.8.2=h22169c7_2
   - libstdcxx-ng=9.1.0=hdf63c60_0
   - libtiff=4.1.0=hfc65ed5_0
@@ -65,12 +82,17 @@ dependencies:
   - matplotlib=3.1.2=py37_0
   - matplotlib-base=3.1.2=py37h250f245_0
   - mdanalysis=0.20.1=py37he1b5a44_0
+  - mistune=0.8.4=py37h516909a_1000
   - mmtf-python=1.1.2=py_0
+  - more-itertools=7.2.0=py_0
   - mpi=1.0=openmpi
   - msgpack-python=0.6.2=py37hc9558a2_0
+  - nbconvert=5.6.1=py37_0
+  - nbformat=4.4.0=py_1
   - ncurses=6.1=hf484d3e_1002
   - netcdf4=1.5.3=py37hf3213d6_0
   - networkx=2.4=py_0
+  - notebook=6.0.1=py37_0
   - nspr=4.23=he1b5a44_0
   - nss=3.47=he751ad9_0
   - numba=0.46.0=py37hb3f55d8_1
@@ -80,17 +102,29 @@ dependencies:
   - openssl=1.1.1d=h516909a_0
   - packaging=19.2=py_0
   - pandas=0.25.3=py37hb3f55d8_0
+  - pandoc=2.7.3=0
+  - pandocfilters=1.4.2=py_1
+  - parso=0.5.1=py_0
   - patsy=0.5.1=py_0
   - pcre=8.43=he1b5a44_0
+  - pexpect=4.7.0=py37_0
+  - pickleshare=0.7.5=py37_1000
   - pillow=6.2.1=py37h6b7be26_0
   - pip=19.3.1=py37_0
+  - prometheus_client=0.7.1=py_0
+  - prompt_toolkit=2.0.10=py_0
   - pthread-stubs=0.4=h14c3975_1001
+  - ptyprocess=0.6.0=py_1001
+  - pygments=2.4.2=py_0
   - pyparsing=2.4.4=py_0
   - pyqt=5.12.3=py37hcca6a23_0
+  - pyrsistent=0.15.5=py37h516909a_0
   - python=3.7.3=h33d41f4_1
   - python-dateutil=2.8.1=py_0
+  - pythreejs=2.1.1=py_0
   - pytz=2019.3=py_0
   - pyyaml=5.1.2=py37h516909a_0
+  - pyzmq=18.1.0=py37h1768529_0
   - qhull=2015.2=h6bb024c_1000
   - qt=5.12.5=hd8c4c69_1
   - readline=8.0=hf8c457e_0
@@ -98,20 +132,30 @@ dependencies:
   - scikit-learn=0.21.3=py37hcdab131_0
   - scipy=1.3.1=py37h921218d_2
   - seaborn=0.9.0=py_1
+  - send2trash=1.5.0=py_0
   - setuptools=41.6.0=py37_1
   - six=1.13.0=py37_0
   - sqlite=3.30.1=hcee41ef_0
   - statsmodels=0.10.1=py37hc1659b7_1
   - tbb=2019.9=hc9558a2_0
+  - terminado=0.8.2=py37_0
+  - testpath=0.4.4=py_0
   - tk=8.6.9=hed695b0_1003
   - tornado=6.0.3=py37h516909a_0
   - tqdm=4.36.1=py_0
+  - traitlets=4.3.3=py37_0
+  - traittypes=0.2.1=py_1
   - umap-learn=0.3.10=py37_0
+  - wcwidth=0.1.7=py_1
+  - webencodings=0.5.1=py_1
   - wheel=0.33.6=py37_0
+  - widgetsnbextension=3.5.1=py37_0
   - xorg-libxau=1.0.9=h14c3975_0
   - xorg-libxdmcp=1.1.3=h516909a_0
   - xz=5.2.4=h14c3975_1001
   - yaml=0.1.7=h14c3975_1001
+  - zeromq=4.3.2=he1b5a44_2
+  - zipp=0.6.0=py_0
   - zlib=1.2.11=h516909a_1006
   - zstd=1.4.3=h3b9ef0a_0
   - pip:

--- a/environment.yml
+++ b/environment.yml
@@ -1,69 +1,95 @@
 name: freud-examples
 channels:
-- conda-forge
-- anaconda
-- defaults
+  - conda-forge
+  - defaults
 dependencies:
-- libgcc-ng=7.2.0=hdf63c60_3
-- libgfortran=3.0.0=1
-- libstdcxx-ng=7.2.0=hdf63c60_3
-- blas=1.1=openblas
-- bokeh=0.13.0=py36_0
-- bzip2=1.0.6=h470a237_2
-- ca-certificates=2018.4.16=0
-- certifi=2018.4.16=py36_0
-- cycler=0.10.0=py_1
-- dbus=1.13.0=h3a4f0e9_0
-- expat=2.2.5=hfc679d8_1
-- fontconfig=2.13.0=hd36ec8e_5
-- freetype=2.8.1=hfa320df_1
-- freud=0.9.0=py36h3e44d54_0
-- gettext=0.19.8.1=0
-- glib=2.55.0=h464dc38_2
-- gst-plugins-base=1.12.5=hde13a9d_0
-- gstreamer=1.12.5=h61a6719_0
-- icu=58.2=hfc679d8_0
-- jinja2=2.10=py_1
-- jpeg=9c=h470a237_0
-- kiwisolver=1.0.1=py36_1
-- libffi=3.2.1=3
-- libiconv=1.15=h470a237_1
-- libpng=1.6.34=ha92aebf_1
-- libtiff=4.0.9=he6b73bb_1
-- libuuid=2.32.1=h470a237_0
-- libxcb=1.13=h470a237_2
-- libxml2=2.9.8=h422b904_2
-- markupsafe=1.0=py36_0
-- matplotlib=2.2.2=py36_1
-- ncurses=6.1=hfc679d8_1
-- numpy=1.15.0=py36_blas_openblashd3ea46f_200
-- olefile=0.45.1=py_1
-- openblas=0.2.20=8
-- openssl=1.0.2o=h470a237_1
-- packaging=17.1=py_0
-- pcre=8.41=h470a237_2
-- pillow=5.2.0=py36h2dc6135_1
-- pip=18.0=py36_1
-- pthread-stubs=0.4=h470a237_1
-- pyparsing=2.2.0=py_1
-- pyqt=5.6.0=py36h8210e8a_6
-- python=3.6.6=h5001a0f_0
-- python-dateutil=2.7.3=py_0
-- pytz=2018.5=py_0
-- pyyaml=3.12=py36_1
-- qt=5.6.2=h50c60fd_8
-- readline=7.0=haf1bffa_1
-- scipy=1.1.0=py36_blas_openblashd3ea46f_201
-- setuptools=40.0.0=py36_1
-- sip=4.18=py36_1
-- six=1.11.0=py36_1
-- sqlite=3.24.0=h2f33b56_0
-- tbb=2018_20171205=0
-- tk=8.6.8=0
-- tornado=5.1=py36h470a237_1
-- wheel=0.31.1=py36_1
-- xorg-libxau=1.0.8=h470a237_6
-- xorg-libxdmcp=1.1.2=h470a237_7
-- xz=5.2.4=h470a237_0
-- yaml=0.1.7=0
-- zlib=1.2.11=h470a237_3
+  - _libgcc_mutex=0.1=main
+  - bokeh=1.3.4=py37_0
+  - bzip2=1.0.8=h516909a_1
+  - ca-certificates=2019.9.11=hecc5488_0
+  - certifi=2019.9.11=py37_0
+  - cycler=0.10.0=py_2
+  - dbus=1.13.6=he372182_0
+  - embree3=3.6.1=hc9558a2_0
+  - expat=2.2.5=he1b5a44_1004
+  - fftw=3.3.8=nompi_h7f3a6c3_1110
+  - fontconfig=2.13.1=h86ecdb6_1001
+  - freetype=2.10.0=he983fc9_1
+  - fresnel=0.11.0=py37h9de70de_0
+  - freud=2.0.0=py37hc9558a2_0
+  - gettext=0.19.8.1=hc5be6a0_1002
+  - glib=2.58.3=h6f030ca_1002
+  - gst-plugins-base=1.14.5=h0935bb2_0
+  - gstreamer=1.14.5=h36ae1b5_0
+  - hoomd=2.8.0=py37h9de70de_0
+  - icu=64.2=he1b5a44_1
+  - jinja2=2.10.3=py_0
+  - joblib=0.14.0=py_0
+  - jpeg=9c=h14c3975_1001
+  - kiwisolver=1.1.0=py37hc9558a2_0
+  - lammps=2019.08.07=py37_openmpi_0
+  - libblas=3.8.0=14_openblas
+  - libcblas=3.8.0=14_openblas
+  - libclang=9.0.0=hc9558a2_1
+  - libffi=3.2.1=he1b5a44_1006
+  - libgcc-ng=9.1.0=hdf63c60_0
+  - libgfortran-ng=7.3.0=hdf63c60_2
+  - libiconv=1.15=h516909a_1005
+  - liblapack=3.8.0=14_openblas
+  - libllvm8=8.0.1=hc9558a2_0
+  - libllvm9=9.0.0=hc9558a2_3
+  - libopenblas=0.3.7=h6e990d7_3
+  - libpng=1.6.37=hed695b0_0
+  - libstdcxx-ng=9.1.0=hdf63c60_0
+  - libtiff=4.1.0=hfc65ed5_0
+  - libuuid=2.32.1=h14c3975_1000
+  - libxcb=1.13=h14c3975_1002
+  - libxkbcommon=0.9.1=hebb1f50_0
+  - libxml2=2.9.10=hee79883_0
+  - llvmlite=0.30.0=py37h8b12597_1
+  - lz4-c=1.8.3=he1b5a44_1001
+  - markupsafe=1.1.1=py37h516909a_0
+  - matplotlib=3.1.2=py37_0
+  - matplotlib-base=3.1.2=py37h250f245_0
+  - mpi=1.0=openmpi
+  - ncurses=6.1=hf484d3e_1002
+  - nspr=4.23=he1b5a44_0
+  - nss=3.47=he751ad9_0
+  - numba=0.46.0=py37hb3f55d8_1
+  - numpy=1.17.3=py37h95a1406_0
+  - olefile=0.46=py_0
+  - openmpi=4.0.1=hc99cbb1_2
+  - openssl=1.1.1d=h516909a_0
+  - packaging=19.2=py_0
+  - pcre=8.43=he1b5a44_0
+  - pillow=6.2.1=py37h6b7be26_0
+  - pip=19.3.1=py37_0
+  - pthread-stubs=0.4=h14c3975_1001
+  - pyparsing=2.4.4=py_0
+  - pyqt=5.12.3=py37hcca6a23_0
+  - python=3.7.3=h33d41f4_1
+  - python-dateutil=2.8.1=py_0
+  - pyyaml=5.1.2=py37h516909a_0
+  - qhull=2015.2=h6bb024c_1000
+  - qt=5.12.5=hd8c4c69_1
+  - readline=8.0=hf8c457e_0
+  - rowan=1.2.2=py_2
+  - scikit-learn=0.21.3=py37hcdab131_0
+  - scipy=1.3.1=py37h921218d_2
+  - setuptools=41.6.0=py37_1
+  - six=1.13.0=py37_0
+  - sqlite=3.30.1=hcee41ef_0
+  - tbb=2019.9=hc9558a2_0
+  - tk=8.6.9=hed695b0_1003
+  - tornado=6.0.3=py37h516909a_0
+  - umap-learn=0.3.10=py37_0
+  - wheel=0.33.6=py37_0
+  - xorg-libxau=1.0.9=h14c3975_0
+  - xorg-libxdmcp=1.1.3=h516909a_0
+  - xz=5.2.4=h14c3975_1001
+  - yaml=0.1.7=h14c3975_1001
+  - zlib=1.2.11=h516909a_1006
+  - zstd=1.4.3=h3b9ef0a_0
+  - pip:
+    - pyqt5-sip==4.19.18
+    - pyqtwebengine==5.12.1

--- a/examples/Calculating Strain via Voxelization.ipynb
+++ b/examples/Calculating Strain via Voxelization.ipynb
@@ -19,14 +19,12 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "%matplotlib inline\n",
     "import freud\n",
     "import numpy as np\n",
     "import matplotlib.pyplot as plt\n",
     "import re\n",
-    "from scipy.sparse import csr_matrix, csc_matrix\n",
-    "%matplotlib inline\n",
-    "from ipywidgets import FloatProgress\n",
-    "from IPython.display import display"
+    "from scipy.sparse import csr_matrix, csc_matrix"
    ]
   },
   {

